src/sanitizer.js

```diff
@@ -8,18 +8,67 @@
 // eslint-disable-next-line no-unused-vars
 export const _sanitizeDocFragment = function _sanitizeDocFragment(config, input) {
     // TODO implement this function and remove eslint-disable comment above
+    // TODO maybe we can use https://developer.mozilla.org/en-US/docs/Web/API/NodeIterator to traverse through nodes
 };
 
 // eslint-disable-next-line no-unused-vars
 export const _sanitizeDocument = function _sanitizeDocument(config, input) {
     // TODO implement this function and remove eslint-disable comment above
 };
 
-// a parser that doesn't run scripts or loads resources
-// eslint-disable-next-line no-unused-vars
+/**
+ * fragmentParser
+ * a parser that doesn't run scripts or loads resources
+ *
+ * @param {Node} ctx DOM node
+ * @param {Object} input Document or DocumentFragment
+ */
+// Q: is the purpose of _fragmentParser is to parse `input` under the context of `ctx`, so that `input` can be append to `ctx` correctly and returned?
 export const _fragmentParser = function _fragmentParser(ctx, input) {
     // note: ctx is of type Node (element?)
-    // TODO implement this function and remove eslint-disable comment above
+    // Reference: https://github.com/mozilla/sanitizer-polyfill/issues/3
+    // https://html.spec.whatwg.org/#concept-frag-parse-context
+    // Q: https://github.com/mozilla/sanitizer-polyfill/issues/3 references we need to call createHTMLDocument, but I don't know why it is necessary yet. Is the intention to attach the result of the parsing `input` to the empty Document we create?
+    
+ 
+    // Create a new Document node, and mark it as being an HTML document.
+    let doc = document.implementation.createHTMLDocument()
+
+    
+    // Check if quirks, no-quirks, or limited-quirks mode
+    // https://dom.spec.whatwg.org/#concept-document-quirks
+    // https://dom.spec.whatwg.org/#dom-document-compatmode
+    if (ctx.ownerDocument.compatMode === "CSS1Compat") {
+        // https://developer.mozilla.org/en-US/docs/Mozilla/Mozilla_quirks_mode_behavior
+    }
+
+
+    // Create a new HTML parser, and associate it with the just created Document node.
+    // TODO there should be documentfragment parser somewhere to reuse here...
+    // Since input is a Document or DocumentFragment, we probably just need to traverse the existing DOM tree?
+
+    // Switching parser state by node type
+    // Q: For the items missing instructions on what state the tokenizer should be (e.g. xmp, style, etc), do we assume the tokenizer should be in data state? https://html.spec.whatwg.org/#concept-frag-parse-context
+    switch (ctx.nodeType) {
```

Comment:
Not sure if this is required? I think the `createContextualFragment` should take care of that, because of the `selectNode` call below?

Reply:
Yup! I thought `input` wasn't a String earlier, so I wasn't sure if we could use `createContextualFragment`. I don't think this block is required anymore too :)